src/arc-login.ts

```diff
@@ -47,52 +51,52 @@ async function getAzureAccessToken(servicePrincipalId, servicePrincipalKey, tena
-            throw new Error('Could not fetch the azure access token: ' + ex);
-        });
+        let method = core.getInput('method');
+        if (method != 'service-account' && method != 'spn'){
+            throw Error("Supported methods for arc cluster are 'service-account' and 'spn'.");
+        }        
+        
         let resourceGroupName = core.getInput('resource-group');
         let clusterName = core.getInput('cluster-name');
-        let saToken = core.getInput('token');
-        return new Promise<string>((resolve, reject) => {
-            var webRequest = new WebRequest();
-            webRequest.method = 'POST';
-            webRequest.uri = `${managementEndpointUrl}/subscriptions/${subscriptionId}/resourceGroups/${resourceGroupName}/providers/Microsoft.Kubernetes/connectedClusters/${clusterName}/listClusterUserCredentials?api-version=2020-01-01-preview`;
-            webRequest.headers = {
-                'Authorization': 'Bearer ' + azureSessionToken,
-                'Content-Type': 'application/json; charset=utf-8'
+        if(!resourceGroupName){
+            throw Error("'resourceGroupName' is not passed for arc cluster.")
+        }
+        if(!clusterName){
+            throw Error("'clusterName' is not passed for arc cluster.")
+        }
+        await az_login.main();
+        await az_login.executeAzCliCommand(`account show`, false);
+        await az_login.executeAzCliCommand(`extension add -n connectedk8s`, false);
+        await az_login.executeAzCliCommand(`extension list`, false);
+        const runnerTempDirectory = process.env['RUNNER_TEMP']; // Using process.env until the core libs are updated
+        const kubeconfigPath = path.join(runnerTempDirectory, `kubeconfig_${Date.now()}`);
+        if (method == 'service-account'){
+            let saToken = core.getInput('token');
+            if(!saToken){
+                throw Error("'saToken' is not passed for 'service-account' method.")
             }
-            webRequest.body = JSON.stringify({
-                authenticationMethod: "Token",
-                value: {
-                    token: saToken
-                }
-            });
-            sendRequest(webRequest).then((response: WebResponse) => {
-                let kubeconfigs = response.body.kubeconfigs;
-                if (kubeconfigs && kubeconfigs.length > 0) {
-                    var kubeconfig = Buffer.from(kubeconfigs[0].value, 'base64');
-                    resolve(kubeconfig.toString());
-                } else {
-                    reject(JSON.stringify(response.body));
-                }
-            }).catch(reject);
-        });
+            console.log('using service account method for authenticating to arc cluster.')
+            spawn('az',['connectedk8s','proxy','-n',clusterName,'-g',resourceGroupName,'-f',kubeconfigPath,'--token',saToken], {
```

Comment:
can we dump the stdout to a file and mention where we are dumping it. So that for debugging purpose the action user can view the CSP logs ?.

Reply:
Lets add an env variable in az proxy itself for logs redirection as we discussed.